src/Microsoft.NetCore.Analyzers/Core/Runtime/TestForEmptyStringsUsingStringLength.Fixer.cs

```diff
@@ -18,12 +25,139 @@ public sealed override FixAllProvider GetFixAllProvider()
             // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
         }
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+
+            SyntaxNode binaryExpressionSyntax = GetBinaryExpression(node);
+
+            if (!IsEqualsOperator(binaryExpressionSyntax) && !IsNotEqualsOperator(binaryExpressionSyntax))
+            {
+                return;
+            }
+
+            SemanticModel model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            FixResolution resolution = TryGetFixResolution(binaryExpressionSyntax, model);
+
+            if (resolution != null)
+            {
+                // We cannot have multiple overlapping diagnostics of this id.
+                Diagnostic diagnostic = context.Diagnostics.Single();
```

Comment:
I would avoid the `Single` invocation here. Instead just pass in `context.Diagnostics` to `RegisterCodeFix` invocation below.

Reply:
sounds good. I was wondering about that, seems like it could be problematic down the road.

Reply:
See https://github.com/dotnet/roslyn-analyzers/pull/1854#discussion_r224774464 for context.